src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/tabs.ts

```diff
@@ -196,35 +203,74 @@ export const updateTabs: InternalStateThunkActionCreator<[TabbedContentState], P
     );
   };
 
-export const updateTabAppStateAndGlobalState: InternalStateThunkActionCreator<[TabActionPayload]> =
-  ({ tabId }) =>
-  (dispatch, _, { runtimeStateManager }) => {
-    dispatch(
-      internalStateSlice.actions.setTabAppStateAndGlobalState({
-        tabId,
-        appState: selectTabRuntimeAppState(runtimeStateManager, tabId),
-        globalState: selectTabRuntimeGlobalState(runtimeStateManager, tabId),
-      })
-    );
-  };
+export const initializeTabs = createInternalStateAsyncThunk(
+  'internalState/initializeTabs',
+  async (
+    {
+      discoverSessionId,
+      shouldClearAllTabs,
+    }: { discoverSessionId: string | undefined; shouldClearAllTabs?: boolean },
+    { dispatch, getState, extra: { services, tabsStorageManager, customizationContext } }
+  ) => {
+    if (TABS_ENABLED && shouldClearAllTabs) {
+      dispatch(clearAllTabs());
+    }
+
+    const { userId: existingUserId, spaceId: existingSpaceId } = getState();
```

Comment:
We now store `userId` and `spaceId` in Redux to dedupe the requests since tabs can be initialized multiple times now.